sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/util/CosmosPagedFlux.java

```diff
@@ -312,14 +313,25 @@ private Flux<FeedResponse<T>> byPage(CosmosPagedFluxOptions pagedFluxOptions, Co
                         CosmosDiagnostics diagnostics = feedResponse != null ?
                             feedResponse.getCosmosDiagnostics() : null;
 
+                        FeedResponseDiagnostics feedResponseDiagnostics = ImplementationBridgeHelpers
+                                .CosmosDiagnosticsHelper
+                                .getCosmosDiagnosticsAccessor()
+                                .getFeedResponseDiagnostics(diagnostics);
+
+                        Duration effectiveLatency = Duration.between(startTime.get(), Instant.now()).minus(
+                                Duration.ofNanos(feedResponseConsumerLatencyInNanos.get()));
+
+                        // record effective latency for feed response creation if not recorded before
+                        if (feedResponseDiagnostics.getFeedResponseCreationLatency() == Duration.ZERO) {
```

Comment:
Placing this check here to ensure the latency does not get recorded twice for the same `FeedResponse` instance. A case in point is `CosmosAsyncEncryptionContainer` where an additional subscription occurs within the query-specific methods leading to `ON_NEXT` being signaled twice and therefore the latency being recorded twice.

Reply:
Can you explain a little bit more around this part? the same signal for the same feed response?

Reply:
The `queryItemsHelper` implementation instantiates a `Flux` by invoking `byPage` on an instance of `CosmosPagedFlux`. This instance of `Flux` is wrapped again to return a `CosmosPagedFlux` to the client. The client needs to invoke `byPage` again to iterate through `feedResponses`. This creates an extra subscription, therefore it is possible that the latency is recorded redundantly for the same `feedResponse` (redundant emission of `feedResponse`). One way to check if it is not a redundant recording of latency is by checking if the `feedResponseDiagnostics.getFeedResponseCreationLatency == Duration.ZERO`